setup.cfg

```diff
@@ -43,15 +43,15 @@ console_scripts =
 [options.extras_require]
 doc =
     furo
-    sphinx
-    sphinx-autodoc-typehints
-    sphinxcontrib-autoprogram
+    sphinx~=3.0
+    sphinx-autodoc-typehints~=1.10
+    sphinxcontrib-autoprogram~=0.1.0
 test =
-    filelock
-    pytest
-    pytest-cov
-    pytest-mock
-    pytest-xdist>=1.34
+    filelock~=3.0
+    pytest>=5,<=6
+    pytest-cov~=2.0
+    pytest-mock>=2,<=3
```

Comment:
Any reason you need to upper pin here? I'd rather we upper pin when the new version breaks us. More often tends to not, see https://www.bernat.tech/version-numbers/ and https://snarky.ca/why-i-dont-like-semver/

Reply:
Hum, I have my reservations but sure.